src/MongoDB.Driver.Core/Core/Authentication/GssapiAuthenticator.cs

```diff
@@ -340,9 +328,9 @@ public ISaslStep Transition(SaslConversation conversation, byte[] bytesReceivedF
                 byte[] bytesToSendToServer;
                 try
                 {
-                    _context.Initialize(_servicePrincipalName, bytesReceivedFromServer, out bytesToSendToServer);
+                    bytesToSendToServer = _context.Next(bytesReceivedFromServer);
```

Comment:
can we make similar refactoring in line 340? Since we already have `_servicePrincipalName, _authorizationId` in the _context?

Reply:
GssapiAuthenticator is an existing class (mostly written by Craig Wilson) that encapsulates the SASL conversation steps. As such I'm trying to minimize changes in this code while implementing GSSAPI support on Linux. We can consider further refactorings once we have ensured that SSPI on Windows and GSSAPI on Linux are working as expected.

Reply:
I'm ok with implementing it further, we should just ensure that we won't forget it